--- gfxboot-4.3.8/bin/help2txt.orig	2011-10-25 15:43:56.304202803 -0200
+++ gfxboot-4.3.8/bin/help2txt	2011-10-25 15:44:25.713345966 -0200
@@ -12,7 +12,7 @@ sub body_start;
 sub find_tag;
 sub nospaces;
 
-$opt_product = "openSUSE";
+$opt_product = "Mandriva Linux";
 
 %help_key_rename = (
   'F2' => 'F3',
--- gfxboot-4.3.8/gfxboot.orig	2011-10-25 15:44:03.848239523 -0200
+++ gfxboot-4.3.8/gfxboot	2011-10-25 15:45:01.503520246 -0200
@@ -916,11 +916,17 @@ sub create_vmdk;
 
 my $opt_verbose = 0;
 my $opt_preview = 0;
-my $opt_gfxarchive = "/boot/message";
-my $opt_bootloader;
-my $opt_vm = "qemu64";
++my $opt_gfxarchive = "/usr/share/gfxboot/themes/Mandriva/boot/message";
++my $opt_bootloader = "grub";
+my $opt_vm = "qemu";
 my $opt_savetemp = 0;
 my $opt_grub = "/";
+my $arch_vendor = `rpm --eval '\%_arch-\%_vendor'`;
+chomp($arch_vendor);
+my $opt_grub_fat_stage1_5 = "$opt_grub/lib/grub/$arch_vendor/fat_stage1_5";
+my $opt_grub_stage1 = "$opt_grub/lib/grub/$arch_vendor/stage1";
+my $opt_grub_stage2 = "$opt_grub/lib/grub/$arch_vendor/stage2";
+my $opt_grub_menu = "/boot/grub/menu.lst";
 my $opt_lilo = "/";
 my $opt_syslinux = "/";
 my $opt_password = undef;
@@ -987,7 +993,7 @@ my %vm_list = (
 my @vm_order = qw ( qemu64 qemu32 qemu qemu-kvm vbox vbox64 vboxsdl vmplayer vmware bochs );
 
 my %bl_list = (
-  grub     => '/usr/sbin/grub',
+  grub     => '/sbin/grub',
   lilo     => '/sbin/lilo',
   isolinux => '/usr/share/syslinux/isolinux.bin',
   syslinux => '/usr/bin/syslinux',
@@ -1010,6 +1016,7 @@ GetOptions(
   'bootloader|b=s'      => \$opt_bootloader,
   'vm|m=s'              => \$opt_vm,
   'grub=s'              => \$opt_grub,
+  'grub-menu=s'         => \$opt_grub_menu,
   'lilo=s'              => \$opt_lilo,
   'isolinux=s'          => \$opt_syslinux,
   'syslinux=s'          => \$opt_syslinux,
@@ -1138,8 +1145,8 @@ if(@opt_extractfiles) {
 }
 
 if($opt_theme) {
-  $theme_dir = "/etc/bootsplash/themes/$opt_theme/bootloader";
-  $theme_archive = "$theme_dir/message";
+  $theme_dir = "/usr/share/gfxboot/themes/$opt_theme";
+  $theme_archive = "$theme_dir/boot/message";
   die "$opt_theme: no such theme\n" unless -f $theme_archive;
   $write_archive = 1;
   ( $work_dir, $work_archive_name ) = unpack_archive $theme_archive unless $work_dir;
@@ -1323,6 +1330,7 @@ Preview/test gfxboot setup:
                                 Add files to test directory.
       --test-rm-files FILE1 FILE2 ...
                                 Delete files from test directory.
+      --grub-menu FILE          Use specified grub menu.lst file.
 
 Adding/removing files from gfxboot archive:
 
@@ -1503,7 +1511,7 @@ sub read_grub_menu
   local $_;
   my ($menu, $default);
 
-  print STDERR "/boot/grub/menu.lst: $!\n" unless open ML, "${sudo}cat /boot/grub/menu.lst 2>/dev/null |";
+  print STDERR "$opt_grub_menu: $!\n" unless open ML, "${sudo}cat $opt_grub_menu 2>/dev/null |";
 
   while(<ML>) {
     push @{$menu->{list}}, $1 if /^\s*title\s+(.+?)\s*$/;
@@ -1581,7 +1589,15 @@ sub prepare_grub
 
   die "Can't setup grub on $opt_media.\n" if $opt_media && $opt_media ne 'disk';
 
-  $opt_grub = unpack_rpm $opt_grub if -f $opt_grub;
+  if (-f $opt_grub) {
+    $opt_grub_fat_stage1_5 = `rpm -qpl $opt_grub | grep /fat_stage1_5\$`;
+    chomp($opt_grub_fat_stage1_5);
+    $opt_grub_stage1 = `rpm -qpl $opt_grub | grep /stage1\$`;
+    chomp($opt_grub_stage1);
+    $opt_grub_stage2 = `rpm -qpl $opt_grub | grep /stage2\$`;
+    chomp($opt_grub_stage2);
+    $opt_grub = unpack_rpm $opt_grub;
+  }
 
   die "error: grub not found\n" unless -x "$opt_grub/$bl_list{grub}";
 
@@ -1600,7 +1616,7 @@ sub prepare_grub
   mkdir "$dst/boot", 0755;
   mkdir "$dst/boot/grub", 0755;
 
-  system "cp $opt_grub/usr/lib/grub/{fat_stage1_5,stage1,stage2} $dst/boot/grub" and die "error: no grub\n";
+  system "cp $opt_grub_fat_stage1_5 $opt_grub_stage1 $opt_grub_stage2 $dst/boot/grub" and die "error: no grub\n";
   system "cp $file $dst/boot/message";
 
   system "cp /boot/vmlinuz $dst/boot" if -f "/boot/vmlinuz";
@@ -1614,7 +1630,7 @@ sub prepare_grub
   print F "default $menu->{default}\ntimeout 20\ngfxmenu (hd0,0)/boot/message\n\n";
 
   for (@{$menu->{list}}) {
-    print F "title $_\n  root (hd0,0)\n  kernel /boot/vmlinuz\n  initrd /boot/initrd\n\n"
+    print F "title $_\n  root (hd0,0)\n  kernel /boot/vmlinuz title=\"$_\"\n  initrd /boot/initrd\n\n"
   }
   close F;
 
@@ -1643,7 +1659,7 @@ sub prepare_grub
 
   my $log = $gfxboot_tmp->file('grub.log');
 
-  open F, "| $opt_grub/usr/sbin/grub --batch --config-file=$dst/boot/grub/menu.lst --device-map=$dst/boot/grub/device.map >$log 2>&1";
+  open F, "| $opt_grub/sbin/grub --batch --config-file=$dst/boot/grub/menu.lst --device-map=$dst/boot/grub/device.map >$log 2>&1";
   print F "setup --prefix=/boot/grub (hd0,0) (hd0,0)\n";
   close F;
 
@@ -2126,7 +2142,7 @@ sub run_qemu
   my $vm_env = shift;
   my $q = $vm_list{$opt_vm}{cmd};
 
-  $q = "MALLOC_CHECK_=0 $q -enable-kvm" if -d "/sys/devices/system/kvm";
+  #$q = "MALLOC_CHECK_=0 $q -enable-kvm" if -d "/sys/devices/system/kvm";
 
   $q .= " -m 512";
 
--- gfxboot-4.3.8/gfxtest.orig	2011-10-25 15:44:14.545291597 -0200
+++ gfxboot-4.3.8/gfxtest	2011-10-25 15:44:25.714345971 -0200
@@ -143,7 +143,7 @@ fi
 
 gfxboot_x_args="$args $@"
 
-opt_theme=${opt_theme:-openSUSE}
+opt_theme=${opt_theme:-Mandriva}
 if [ ! -d "themes/$opt_theme" ] ; then
   echo "no such theme: $opt_theme"
   exit 1
