--- Makefile.link	2010-12-20 09:11:25.000000000 +0000
+++ Makefile	2010-12-20 09:11:51.000000000 +0000
@@ -10,7 +10,7 @@
 all:	bin2c gfxboot-compile bincode gfxboot-font addblack
 
 gfxboot-font: gfxboot-font.c
-	$(CC) $(CFLAGS) -I /usr/include/freetype2 -lfreetype $< -o $@
+	$(CC) $(CFLAGS) -I /usr/include/freetype2 $< -o $@ -lfreetype
 
 gfxboot-compile: gfxboot-compile.c vocabulary.h bincode.h
 	$(CC) $(CFLAGS) $< -o $@
