--- gfxboot-4.5.103/gfxboot
+++ gfxboot-4.5.103/gfxboot
@@ -995,9 +995,9 @@
 my %bl_list = (
   grub     => '/sbin/grub',
   lilo     => '/sbin/lilo',
-  isolinux => '/usr/share/syslinux/isolinux.bin',
+  isolinux => '/usr/lib/syslinux/isolinux.bin',
   syslinux => '/usr/bin/syslinux',
-  pxelinux => '/usr/share/syslinux/pxelinux.0',
+  pxelinux => '/usr/lib/syslinux/pxelinux.0',
   bd       => '/usr/bin/bd',
   bochs    => '/usr/bin/bochs',
 );
@@ -1809,8 +1809,8 @@
   $arch_dir = 'i386';
   $arch_dir = 'x86_64' if $opt_64 && !$opt_32;
 
-  $comboot = "$opt_syslinux/usr/share/syslinux/gfxboot.c32";
-  $comboot = "$opt_syslinux/usr/share/syslinux/gfxboot.com" unless -f $comboot;
+  $comboot = "$opt_syslinux/usr/lib/syslinux/gfxboot.c32";
+  $comboot = "$opt_syslinux/usr/lib/syslinux/gfxboot.com" unless -f $comboot;
   $comboot = 0 unless -f $comboot;
 
   # syslinux 6.x
@@ -1883,7 +1883,7 @@
     close F;
   }
 
-  system "cp $opt_syslinux/usr/share/syslinux/isolinux.bin $dst/$loader" and die "error: no isolinux\n";
+  system "cp $opt_syslinux/usr/lib/syslinux/isolinux.bin $dst/$loader" and die "error: no isolinux\n";
   system "cp $comboot $dst/$loader" if $comboot;
 
   for my $f ("ldlinux.c32", "libcom32.c32") {
@@ -1933,8 +1933,8 @@
 
   die "error: syslinux not found\n" unless -f "$opt_syslinux/$bl_list{syslinux}";
 
-  $comboot = "$opt_syslinux/usr/share/syslinux/gfxboot.c32";
-  $comboot = "$opt_syslinux/usr/share/syslinux/gfxboot.com" unless -f $comboot;
+  $comboot = "$opt_syslinux/usr/lib/syslinux/gfxboot.c32";
+  $comboot = "$opt_syslinux/usr/lib/syslinux/gfxboot.com" unless -f $comboot;
   $comboot = 0 unless -f $comboot;
 
   # syslinux 6.x
@@ -2050,8 +2050,8 @@
   $arch_dir = 'i386';
   $arch_dir = 'x86_64' if $opt_64 && !$opt_32;
 
-  $comboot = "$opt_syslinux/usr/share/syslinux/gfxboot.c32";
-  $comboot = "$opt_syslinux/usr/share/syslinux/gfxboot.com" unless -f $comboot;
+  $comboot = "$opt_syslinux/usr/lib/syslinux/gfxboot.c32";
+  $comboot = "$opt_syslinux/usr/lib/syslinux/gfxboot.com" unless -f $comboot;
   $comboot = 0 unless -f $comboot;
 
   # syslinux 6.x
@@ -2124,7 +2124,7 @@
     close F;
   }
 
-  system "cp $opt_syslinux/usr/share/syslinux/pxelinux.0 $dst/$loader" and die "error: no pxelinux\n";
+  system "cp $opt_syslinux/usr/lib/syslinux/pxelinux.0 $dst/$loader" and die "error: no pxelinux\n";
   system "cp $comboot $dst/$loader" if $comboot;
 
   for my $f ("ldlinux.c32", "libcom32.c32") {
