--- gfxboot-4.1.19/gfxboot.orig	2011-06-08 15:10:02.311932354 -0300
+++ gfxboot-4.1.19/gfxboot	2011-06-08 15:09:08.619932380 -0300
@@ -909,11 +909,17 @@ sub unpack_rpm;
 
 my $opt_verbose = 0;
 my $opt_preview = 0;
-my $opt_gfxarchive = "/boot/message";
-my $opt_bootloader;
+my $opt_gfxarchive = "/usr/share/gfxboot/themes/Mandriva/boot/message";
+my $opt_bootloader = "grub";
 my $opt_vm = "qemu";
 my $opt_savetemp = 0;
 my $opt_grub = "/";
+my $arch_vendor = `rpm --eval '\%_arch-\%_real_vendor'`;
+chomp($arch_vendor);
+my $opt_grub_fat_stage1_5 = "$opt_grub/lib/grub/$arch_vendor/fat_stage1_5";
+my $opt_grub_stage1 = "$opt_grub/lib/grub/$arch_vendor/stage1";
+my $opt_grub_stage2 = "$opt_grub/lib/grub/$arch_vendor/stage2";
+my $opt_grub_menu = "/boot/grub/menu.lst";
 my $opt_lilo = "/";
 my $opt_syslinux = "/";
 my $opt_password = undef;
@@ -970,7 +976,7 @@ my %vm_list = (
 my @vm_order = qw ( qemu vbox vboxsdl vmplayer vmware qemu-i386 qemu-x86_64 bochs );
 
 my %bl_list = (
-  grub     => '/usr/sbin/grub',
+  grub     => '/sbin/grub',
   lilo     => '/sbin/lilo',
   isolinux => '/usr/share/syslinux/isolinux.bin',
   syslinux => '/usr/bin/syslinux',
@@ -991,6 +997,7 @@ GetOptions(
   'bootloader|b=s'      => \$opt_bootloader,
   'vm|m=s'              => \$opt_vm,
   'grub=s'              => \$opt_grub,
+  'grub-menu=s'         => \$opt_grub_menu,
   'lilo=s'              => \$opt_lilo,
   'isolinux=s'          => \$opt_syslinux,
   'syslinux=s'          => \$opt_syslinux,
@@ -1108,8 +1115,8 @@ if(@opt_extractfiles) {
 }
 
 if($opt_theme) {
-  $theme_dir = "/etc/bootsplash/themes/$opt_theme/bootloader";
-  $theme_archive = "$theme_dir/message";
+  $theme_dir = "/usr/share/gfxboot/themes/$opt_theme";
+  $theme_archive = "$theme_dir/boot/message";
   die "$opt_theme: no such theme\n" unless -f $theme_archive;
   $write_archive = 1;
   ( $work_dir, $work_archive_name ) = unpack_archive $theme_archive unless $work_dir;
@@ -1279,6 +1286,7 @@ Preview/test gfxboot setup:
       --disk                    Create harddisk image for preview.
       --floppy                  Create floppy image for preview.
       --save-image FILE         Copy preview image to FILE.
+      --grub-menu FILE          Use specified grub menu.lst file.
 
 Adding/removing files from gfxboot archive:
 
@@ -1439,7 +1447,7 @@ sub read_grub_menu
   local $_;
   my ($menu, $default);
 
-  print STDERR "/boot/grub/menu.lst: $!\n" unless open ML, "${sudo}cat /boot/grub/menu.lst 2>/dev/null |";
+  print STDERR "$opt_grub_menu: $!\n" unless open ML, "${sudo}cat $opt_grub_menu 2>/dev/null |";
 
   while(<ML>) {
     push @{$menu->{list}}, $1 if /^\s*title\s+(.+?)\s*$/;
@@ -1517,7 +1525,15 @@ sub prepare_grub
 
   die "Can't setup grub on $opt_media.\n" if $opt_media && $opt_media ne 'disk';
 
-  $opt_grub = unpack_rpm $opt_grub if -f $opt_grub;
+  if (-f $opt_grub) {
+    $opt_grub_fat_stage1_5 = `rpm -qpl $opt_grub | grep /fat_stage1_5\$`;
+    chomp($opt_grub_fat_stage1_5);
+    $opt_grub_stage1 = `rpm -qpl $opt_grub | grep /stage1\$`;
+    chomp($opt_grub_stage1);
+    $opt_grub_stage2 = `rpm -qpl $opt_grub | grep /stage2\$`;
+    chomp($opt_grub_stage2);
+    $opt_grub = unpack_rpm $opt_grub;
+  }
 
   die "error: grub not found\n" unless -x "$opt_grub/$bl_list{grub}";
 
@@ -1536,7 +1552,7 @@ sub prepare_grub
   mkdir "$dst/boot", 0755;
   mkdir "$dst/boot/grub", 0755;
 
-  system "cp $opt_grub/usr/lib/grub/{fat_stage1_5,stage1,stage2} $dst/boot/grub" and die "error: no grub\n";
+  system "cp $opt_grub_fat_stage1_5 $opt_grub_stage1 $opt_grub_stage2 $dst/boot/grub" and die "error: no grub\n";
   system "cp $file $dst/boot/message";
 
   system "cp /boot/vmlinuz $dst/boot" if -f "/boot/vmlinuz";
@@ -1550,7 +1566,7 @@ sub prepare_grub
   print F "default $menu->{default}\ntimeout 20\ngfxmenu (hd0,0)/boot/message\n\n";
 
   for (@{$menu->{list}}) {
-    print F "title $_\n  root (hd0,0)\n  kernel /boot/vmlinuz\n  initrd /boot/initrd\n\n"
+    print F "title $_\n  root (hd0,0)\n  kernel /boot/vmlinuz title=\"$_\"\n  initrd /boot/initrd\n\n"
   }
   close F;
 
@@ -1570,7 +1586,7 @@ sub prepare_grub
 
   my $log = $gfxboot_tmp->file('grub.log');
 
-  open F, "| $opt_grub/usr/sbin/grub --batch --config-file=$dst/boot/grub/menu.lst --device-map=$dst/boot/grub/device.map >$log 2>&1";
+  open F, "| $opt_grub/sbin/grub --batch --config-file=$dst/boot/grub/menu.lst --device-map=$dst/boot/grub/device.map >$log 2>&1";
   print F "setup --prefix=/boot/grub (hd0,0) (hd0,0)\n";
   close F;
 
